From ff0ca9ed2bdba08d9113a12064900225919bb522 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Mon, 6 May 2019 21:23:50 -0600
Subject: [PATCH 2/2] [SQUASH][DNM] Merge tag 'android-9.0.0_r37' into
 staging/lineage-16.0_merge-android-9.0.0_r37

Android 9.0.0 Release 37 (PQ3A.190505.002)

* tag 'android-9.0.0_r37':
  Allow lmkd to renice process before killing
  Allow init to set powerctl property

https://github.com/haggertk/android_system_sepolicy/tree/lineage-16.0_merge-android-9.0.0_r37

Change-Id: I2b9b849b1fe4bd162ac32dbb108b66c659623fbe
---
 prebuilts/api/28.0/private/init.te | 3 +++
 prebuilts/api/28.0/public/lmkd.te  | 4 ++--
 private/init.te                    | 3 +++
 public/lmkd.te                     | 4 ++--
 4 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/prebuilts/api/28.0/private/init.te b/prebuilts/api/28.0/private/init.te
index e9959d3d..8ba050fa 100644
--- a/prebuilts/api/28.0/private/init.te
+++ b/prebuilts/api/28.0/private/init.te
@@ -20,3 +20,6 @@ domain_trans(init, { rootfs toolbox_exec }, modprobe)
 userdebug_or_eng(`
   domain_auto_trans(init, logcat_exec, logpersist)
 ')
+
+# Allow the BoringSSL self test to request a reboot upon failure
+set_prop(init, powerctl_prop)
diff --git a/prebuilts/api/28.0/public/lmkd.te b/prebuilts/api/28.0/public/lmkd.te
index 472946ec..5b4a235a 100644
--- a/prebuilts/api/28.0/public/lmkd.te
+++ b/prebuilts/api/28.0/public/lmkd.te
@@ -21,8 +21,8 @@ allow lmkd system_server:file write;
 r_dir_file(lmkd, sysfs_lowmemorykiller)
 allow lmkd sysfs_lowmemorykiller:file w_file_perms;
 
-# Send kill signals
-allow lmkd appdomain:process sigkill;
+# setsched and send kill signals
+allow lmkd appdomain:process { setsched sigkill };
 
 # Clean up old cgroups
 allow lmkd cgroup:dir { remove_name rmdir };
diff --git a/private/init.te b/private/init.te
index e9959d3d..8ba050fa 100644
--- a/private/init.te
+++ b/private/init.te
@@ -20,3 +20,6 @@ domain_trans(init, { rootfs toolbox_exec }, modprobe)
 userdebug_or_eng(`
   domain_auto_trans(init, logcat_exec, logpersist)
 ')
+
+# Allow the BoringSSL self test to request a reboot upon failure
+set_prop(init, powerctl_prop)
diff --git a/public/lmkd.te b/public/lmkd.te
index 472946ec..5b4a235a 100644
--- a/public/lmkd.te
+++ b/public/lmkd.te
@@ -21,8 +21,8 @@ allow lmkd system_server:file write;
 r_dir_file(lmkd, sysfs_lowmemorykiller)
 allow lmkd sysfs_lowmemorykiller:file w_file_perms;
 
-# Send kill signals
-allow lmkd appdomain:process sigkill;
+# setsched and send kill signals
+allow lmkd appdomain:process { setsched sigkill };
 
 # Clean up old cgroups
 allow lmkd cgroup:dir { remove_name rmdir };
-- 
2.17.1

